lib/pbench/test/unit/server/test_user_management_cli.py

```diff
@@ -0,0 +1,236 @@
+import datetime
+
+import pytest
+from click.testing import CliRunner
+
+from pbench.cli.server.user_management import (
+    USER_LIST_HEADER_ROW,
+    USER_LIST_ROW_FORMAT,
+    user_create,
+    user_delete,
+    user_list,
+    user_update,
+)
+from pbench.server.database.models.users import User
+
+
+def create_user():
+    user = User(
+        username=TestUserManagement.USER_TEXT,
+        password=TestUserManagement.PSWD_TEXT,
+        first_name=TestUserManagement.FIRST_NAME_TEXT,
+        last_name=TestUserManagement.LAST_NAME_TEXT,
+        email=TestUserManagement.EMAIL_TEXT,
+        registered_on=datetime.datetime.now(),
+    )
+    return user
+
+
+def mock_valid_list():
+    user = create_user()
+    return [user]
+
+
+def mock_valid_delete(**kwargs):
+    return
+
+
+def mock_valid_query(**kwargs):
+    return create_user()
+
+
+class TestUserManagement:
+    USER_SWITCH = "--username"
+    PSWD_SWITCH = "--password"
+    PSWD_PROMPT = "Password: "
+    EMAIL_SWITCH = "--email"
+    FIRST_NAME_SWITCH = "--first-name"
+    LAST_NAME_SWITCH = "--last-name"
+    ROLE_SWITCH = "--role"
+    USER_TEXT = "test_user"
+    PSWD_TEXT = "password"
+    EMAIL_TEXT = "test@domain.com"
+    FIRST_NAME_TEXT = "First"
+    LAST_NAME_TEXT = "Last"
+
+    @staticmethod
+    def test_help(pytestconfig):
+        runner = CliRunner(mix_stderr=False)
+        result = runner.invoke(user_create, ["--help"])
+        assert result.exit_code == 0, result.stderr
+        assert str(result.stdout).startswith("Usage:")
+
+    @staticmethod
+    def test_valid_user_registration_with_password_input(
+        db_session, server_config, pytestconfig
```

Comment:
What does it mean to have the `db_session` fixture, here?

I believe that we _don't want_ the CUT to access the DB -- instead, all accesses to it should hit mocks -- so we shouldn't need to have a DB session created, instead we should be referencing a fixture which installs mocks for all the methods.

Ideally, there should be some way for the end of the test to assert that only certain DB methods (or none) were called.

Is this coming in a follow-on PR?

Reply:
Yes for this test we dont need a `db_session` fixture, but just to be clear, mocked db methods are used only for testing  inter-command dependencies for example (user_delete, update and list). The user_create command tests which does not need any previous data to deal with is kept with using in-memory database session